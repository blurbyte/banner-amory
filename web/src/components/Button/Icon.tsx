import styled from 'styled-components';

const Icon = styled.div`
  position: absolute;
  bottom: -1rem;
  left: -2rem;
`;

export default Icon;
