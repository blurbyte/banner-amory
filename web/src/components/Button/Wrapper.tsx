import styled from 'styled-components';
import { transparentize } from 'polished';

import { Color } from '@styles/color';

const Wrapper = styled.button`
  position: relative;
  display: flex;
  justify-content: flex-end;
  width: 100%;
  height: 5rem;
  width: 15rem;
  background: ${transparentize(0.9, Color.Blue)};
  padding: 1.4rem 1rem 0 1rem;
  border: 0;
  outline: 0;
  border-top: 4px solid transparent;
  cursor: pointer;
  font-size: 1.8rem;
  font-weight: 700;
  user-select: none;
  transition: all 0.15s linear;

  &:focus {
    border-top: 4px solid ${Color.Blue};
  }
`;

export default Wrapper;
