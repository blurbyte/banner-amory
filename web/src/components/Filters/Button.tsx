import styled from 'styled-components';
import { transparentize } from 'polished';

import { Colors } from '@styles/colors';

type ButtonProps = {
  isExpanded?: boolean;
};

const Button = styled.button<ButtonProps>`
  color: ${props => (props.isExpanded ? Colors.white : transparentize(0.9, Colors.blue))};
  padding: 0;
  padding-bottom: 0.8rem;
  display: flex;
  justify-content: center;
  align-items: flex-end;
  width: 10rem;
  height: 2rem;
  border: none;
  cursor: pointer;
  outline: none;
  transform: ${props => (props.isExpanded ? 'rotateX(180deg)' : 'rotateX(0)')};
  transform-origin: top center;
  transition: all 0.4s ease-in-out;
`;

export default Button;
