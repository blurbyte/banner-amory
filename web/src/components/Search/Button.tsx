import * as React from 'react';
import styled from 'styled-components';

import { Magnifier } from '../Icons';

const Hitbox = styled.button`
  border: none;
  background: transparent;
  position: absolute;
  outline: none;
  padding: 0;
  cursor: pointer;
  top: 0.8rem;
  right: 0.8rem;
`;

class Button extends React.Component {
  render() {
    return (
      <Hitbox type="submit">
        <Magnifier />
      </Hitbox>
    );
  }
}

export default Button;
