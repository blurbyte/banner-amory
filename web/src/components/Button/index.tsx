import * as React from 'react';
import Wrapper from './Wrapper';
import Icon from './Icon';

type ButtonProps = {
  children: React.ReactNode;
  icon?: React.ComponentType<any>;
  type?: string;
}

class Button extends React.Component<ButtonProps> {
  render() {
    const { children, icon: InjectedIcon, type } = this.props;
    return (
    <Wrapper type={type}>{InjectedIcon && <Icon><InjectedIcon /></Icon>}{children}</Wrapper>
    );
  }
}

export default Button;
