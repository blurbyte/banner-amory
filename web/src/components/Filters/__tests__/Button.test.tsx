import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import Button from '../Button';

test('renders correctly', () => {
  const component = renderer.create(<Button />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('renders correctly when filters are expanded', () => {
  const component = renderer.create(<Button isExpanded />);
  expect(component.toJSON()).toMatchSnapshot();
});
