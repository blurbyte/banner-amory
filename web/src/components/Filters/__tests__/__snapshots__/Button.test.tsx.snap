// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  color: rgba(61,38,218,0.1);
  padding: 0;
  padding-bottom: 0.8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  width: 10rem;
  height: 2rem;
  border: none;
  cursor: pointer;
  outline: none;
  -webkit-transform: rotateX(0);
  -ms-transform: rotateX(0);
  transform: rotateX(0);
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
  -webkit-transition: all 0.4s ease-in-out;
  transition: all 0.4s ease-in-out;
}

<button
  className="c0"
/>
`;

exports[`renders correctly when filters are expanded 1`] = `
.c0 {
  color: #ffffff;
  padding: 0;
  padding-bottom: 0.8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  width: 10rem;
  height: 2rem;
  border: none;
  cursor: pointer;
  outline: none;
  -webkit-transform: rotateX(180deg);
  -ms-transform: rotateX(180deg);
  transform: rotateX(180deg);
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
  -webkit-transition: all 0.4s ease-in-out;
  transition: all 0.4s ease-in-out;
}

<button
  className="c0"
/>
`;
